### ISSUE TITLE
AMP Issue?

### ISSUE THREAD

**Issue by geoffmckay** [kraftbj/genesis-enews-extended#154]
We use this plugin on an WP site.
When AMP is enabled, the Subscribe/Submit generates an error.
If AMP is disabled, it works fine.

Does anyone know how to resolve this issue?

DETAILS:
WP Theme: Genesis Ver. 3.3.3
Genesis Child Theme: Navigation Pro Ver. 1.2.0

Plugins:
Genesis eNews Extended Ver. 2.2.0
AMP Ver. 2.1.2

Email List Manager: Mailchimp

The Plugin works great when AMP is disabled. The issue only happens when AMP is enabled.
The Submit/Subscribe connects with the Mailchimp List, and it adds the subscribe to the MC List.
However, the redirect to the Mailchimp Subscriber Success Notice Page fails.

ERROR DETAILS:
When AMP is enabled, the re-direct to the success page for the subscribe fails. Instead, an Empty Error Code is displayed.

The following is displayed in a red box in the subscribe widget area:
“Your submission failed. The server responded with (code). Please contact the developer of this form processor to improve this message. Learn More”

There is no error code provided. Just the word “code”.
The “Learn More” in the error displayed links to:
https://amp-wp.org/documentation/playbooks/handling-form-submissions/

Using Google Chrome Inspect, the following errors are detected:
__amp_source_origin
from origin ‘…’
has been blocked by CORS policy: No ‘Access-Control-Allow-Origin’ header is present on the requested resource. If an opaque response serves your needs, set the request’s mode to ‘no-cors’ to fetch the resource with CORS disabled.

As mentioned, the Genesis eNews Extended plugin does process the “Subscribe” with the Mailchimp List, but the website shows an Error (with empty error code), instead of re-direct to the success page on Mailchimp.

The site is not in production yet, so I can not provide a public link. 

However, here is the error displayed when you attempt to subscribe:
![Capture](https://user-images.githubusercontent.com/25710480/120080192-b8e30080-c085-11eb-8bf6-9044be6d4acb.PNG)

Do you have any suggestions or ideas?
Thanks so much.
Geoff

**Comment by kraftbj** [comment]
Closing as stale — no specific repro was provided and it's been ~5 years. The widget does include AMP-aware handling of `target="_blank"` (to accommodate amp-form's redirect constraints). If you're still seeing an AMP-specific problem, please open a new issue with the page URL and a description of the failure mode.
